android/app/src/main/kotlin/com/jabook/app/jabook/compose/data/worker/ChapterDetectionWorker.kt

```
@@ -0,0 +1,174 @@
+        loggerFactory: LoggerFactory,
+    ) : CoroutineWorker(appContext, params) {
+        private val logger = loggerFactory.get("ChapterDetectionWorker")
+
+        override suspend fun doWork(): Result =
+            withContext(Dispatchers.IO) {
+                val bookId = inputData.getString(KEY_BOOK_ID).orEmpty()
+                val filePath = inputData.getString(KEY_FILE_PATH).orEmpty()
+                val fileIndex = inputData.getInt(KEY_FILE_INDEX, 0)
+                val totalDurationMs = inputData.getLong(KEY_DURATION_MS, 0L).coerceAtLeast(0L)
+
+                if (bookId.isBlank() || filePath.isBlank() || totalDurationMs <= 0L || fileIndex < 0) {
```

detekt (1.23.8)

[warning] This condition is too complex (4). Defined complexity threshold for conditions is set to '4'

(detekt.complexity.ComplexCondition)